src/pyfaf/actions/create_problems.py

```diff
@@ -341,66 +350,145 @@ def _create_problems(self, db, problemplugin, report_min_count=0):
-                    problem_last_occurrence < db_report.last_occurrence):
-                    problem_last_occurrence = db_report.last_occurrence
-
-                if (problem_first_occurrence is None or
-                    problem_first_occurrence > db_report.first_occurrence):
-                    problem_first_occurrence = db_report.first_occurrence
-
-                if db_report.component not in comps:
-                    comps[db_report.component] = 0
-
-                comps[db_report.component] += 1
-
-            # In case nothing changed, we don't want to mark db_problem dirty
-            # which would cause another UPDATE
-            if db_problem.first_occurrence != problem_first_occurrence:
-                db_problem.first_occurrence = problem_first_occurrence
-            if db_problem.last_occurrence != problem_last_occurrence:
-                db_problem.last_occurrence = problem_last_occurrence
-
-            if reports_changed:
-                db_comps = sorted(comps, key=lambda x: comps[x], reverse=True)
-
-                order = 0
-                for db_component in db_comps:
-                    order += 1
-
-                    db_pcomp = get_problem_component(db, db_problem, db_component)
-                    if db_pcomp is None:
-                        db_pcomp = ProblemComponent()
-                        db_pcomp.problem = db_problem
-                        db_pcomp.component = db_component
-                        db_pcomp.order = order
-                        db.session.add(db_pcomp)
-
-        self.log_debug("Removing {0} invalid reports from problems"
-                       .format(len(invalid_report_ids_to_clean)))
-        for report_id in invalid_report_ids_to_clean:
-            db_report = get_report_by_id(db, report_id)
-            if db_report is not None:
-                db_report.problem_id = None
-                db.session.add(db_report)
-
-        if report_min_count > 0:
-            self.log_debug("Removing problems form low count reports")
-            remove_problem_from_low_count_reports_by_type(db, problemplugin.name,
-                min_count=report_min_count)
-
-        self.log_debug("Flushing session")
-        db.session.flush()
+        if speedup:
+            for problem in problems:
+                if len(problem) <= 0:
+                    continue
+                first_report = next(iter(problem))
+                if len(problem) > 1:
+                    # Find assigned report
+                    origin_report = None
+                    for db_report in problem:
+                        if db_report.problem_id:
+                            origin_report = db_report
+
+                    # Problem created only from new reports
+                    comps = {}
+                    if not origin_report:
+                        new = Problem()
+                        db.session.add(new)
+                        db.session.flush()
```

Comment:
why is flush() here. I am asking because on line 423 you are flushing it again, which makes more sense to me.